### PR TITLE
Add doxygen comments

### PR DESCRIPTION
**Pull request by bmagyar**
Pirated the doxygen content from https://github.com/ros-controls/ros2_control/pull/728

### CODE REVIEW

File: controller_interface/include/controller_interface/chainable_controller_interface.hpp [hunk 1]
```diff
@@ -40,6 +40,13 @@ class ChainableControllerInterface : public ControllerInterfaceBase
   CONTROLLER_INTERFACE_PUBLIC
   virtual ~ChainableControllerInterface() = default;
 
+  /**
+   * This is the method called in the control loop.
```
Comment (destogl) [inline controller_interface/include/controller_interface/chainable_controller_interface.hpp:44]:
```suggestion
   * Control step update. Command interfaces are updated based on on reference inputs and current states.
   * **The method called in the (real-time) control loop.**
```

File: controller_interface/include/controller_interface/controller_interface_base.hpp [hunk 2]
```diff
@@ -105,6 +105,13 @@ class ControllerInterfaceBase : public rclcpp_lifecycle::node_interfaces::Lifecy
   CONTROLLER_INTERFACE_PUBLIC
   virtual CallbackReturn on_init() = 0;
 
+  /**
+   * This is the method called in the control loop.
```
Comment (destogl) [inline controller_interface/include/controller_interface/controller_interface_base.hpp:109]:
```suggestion
   * Control step update. Command interfaces are updated based on on reference inputs and current states.
   * **The method called in the (real-time) control loop.**
```

File: controller_manager/include/controller_manager/controller_manager.hpp [hunk 3]
```diff
@@ -137,13 +137,31 @@ class ControllerManager : public rclcpp::Node
     bool activate_asap = kWaitForAllResources,
     const rclcpp::Duration & timeout = rclcpp::Duration::from_nanoseconds(kInfiniteTimeout));
 
+  /**
+   * \brief      Read from state interfaces
```
Comment (destogl) [inline controller_manager/include/controller_manager/controller_manager.hpp:141]:
```suggestion
  /// Read values to state interfaces.
  /**
   * Read current values from hardware to state interfaces.
   * **The method called in the (real-time) control loop.**
```

File: controller_manager/include/controller_manager/controller_manager.hpp [hunk 4]
```diff
@@ -137,13 +137,31 @@ class ControllerManager : public rclcpp::Node
...
+   *
+   * \param[in]  time    The time at the start of this control loop iteration
+   * \param[in]  period  The measured period of the last control loop iteration
+   */
   CONTROLLER_MANAGER_PUBLIC
   void read(const rclcpp::Time & time, const rclcpp::Duration & period);
 
+  /**
+   * \brief      Run update on controllers
```
Comment (destogl) [inline controller_manager/include/controller_manager/controller_manager.hpp:150]:
```suggestion
  /// Run update on controllers
  /**
   * Call update of all controllers.      
   * **The method called in the (real-time) control loop.**
```

File: controller_manager/include/controller_manager/controller_manager.hpp [hunk 5]
```diff
@@ -137,13 +137,31 @@ class ControllerManager : public rclcpp::Node
...
+   *
+   * \param[in]  time    The time at the start of this control loop iteration
+   * \param[in]  period  The measured period of the last control loop iteration
+   */
   CONTROLLER_MANAGER_PUBLIC
   controller_interface::return_type update(
     const rclcpp::Time & time, const rclcpp::Duration & period);
 
+  /**
+   * \brief      Write to control interfaces
```
Comment (destogl) [inline controller_manager/include/controller_manager/controller_manager.hpp:160]:
```suggestion
  /// Write values from command interfaces.
  /**
   * Write values from command interface into hardware.
   * **The method called in the (real-time) control loop.**
```

**Review by destogl — CHANGES_REQUESTED** [review]

**Comment by bmagyar** [comment]
Sehr gut suggestions

File: controller_manager/include/controller_manager/controller_manager.hpp [hunk 6]
```diff
@@ -137,13 +137,37 @@ class ControllerManager : public rclcpp::Node
...
+   *
+   * \param[in]  time    The time at the start of this control loop iteration
+   * \param[in]  period  The measured period of the last control loop iteration
+   */
   CONTROLLER_MANAGER_PUBLIC
   void read(const rclcpp::Time & time, const rclcpp::Duration & period);
 
+  /// Run update on controllers
+  /**
+   * Call update of all controllers.
```
Comment (bmagyar) [inline controller_manager/include/controller_manager/controller_manager.hpp:153]:
```suggestion
   * Call update of all controllers.
```

**Review by destogl — APPROVED** [review]

**Pull request merged**